src/fragments/lib/storage/flutter/list.mdx

```diff
@@ -1,13 +1,79 @@
-You can list all of the objects uploaded under a given prefix. This will list all public files (i.e. those with `guest` access level):
+You can list all of the files uploaded under a given path.
+
+<BlockSwitcher>
+
+<Block name="Stable (Mobile)">
+
+This will list all public files (i.e. those with `guest` access level):
 
 ```dart
 Future<void> listItems() async {
   try {
     final result = await Amplify.Storage.list();
     final items = result.items;
-    print('Got items: $items');
+    safePrint('Got items: $items');
   } on StorageException catch (e) {
-    print('Error listing items: $e');
+    safePrint('Error listing items: $e');
+  }
+}
+```
+
+</Block>
+
+<Block name = "Developer Preview (Mobile, Web & Desktop)">
+
+This will list all files located under path `album` that:
+* files are with `private` access level
+* list result doesn't include files under any sub path
+* list with pagination, each page contain 50 items
+
+```dart
+
+String? nextToken;
+
+Future<void> listAlbum() async {
+  try {
+    final result = await Amplify.Storage.list(
+      path: 'album/',
+      options: const S3ListOptions(
+        accessLevel: StorageAccessLevel.private,
+        pageSize: 50,
+        excludeSubPaths: true,
+      ),
+    ).result;
+
+    if (result.hasNextPage) {
+      nextToken = result.nextToken;
+    }
+
+    safePrint('Listed items: ${result.items}');
+  } on S3Exception catch (e) {
+    safePrint(e.message);
+  }
+}
+
+```
+
+Pagination is enabled by default. The default `pageSize` is `1000` if it has not been defined in the `S3ListOptions`. You can also list all files under a given path without pagination.
```

Comment:
listing all files requires that you use the ALL option, right?

Reply:
Yes! Sorry this part has been updated including a clearer instruction to use `listAll` option.